### DISCUSSION TITLE
Implementing Efficient Storage and Search Mechanisms for User Information and Multimedia Files Using Milvus and Langchain

### DISCUSSION THREAD

**Discussion by arian360** [milvus-io/milvus#33446] [Q&A and General discussion]
Dear Team,

I am currently working on a project where we need to store user information such as education, employment, social preferences, and various user documents, including images, PDFs, audio, and video files.

A key requirement is to implement a storage solution and a comprehensive search mechanism that allows users to search for any type of information related to their profile, including both textual data and multimedia files. The system should be capable of searching for similar content and retrieving relevant data efficiently.

I am seeking your expertise to determine the best approach to implement this search mechanism using Milvus and Langchain.

Thank you.
arian360

**Comment by yhmo** [comment]
Basically, Milvus is an in-memory vector database. To conduct a search on a collection, it requires the collection's data to be loaded into memory. So, we don't recommend storing images/PDF/video files in Milvus.

A possible solution: store images/pfd/video files in a storage service such as S3, and store embeddings and metadata in Milvus.
For example, we create a collection in Milvus with these fields: ID, embedding, file_path, and the "file_path" stores the paths of the files on S3. After search, you get the file_path from the results, and fetch the files from S3.

**Comment by xiaofan-luan** [comment]
> Dear Team,
> 
> I am currently working on a project where we need to store user information such as education, employment, social preferences, and various user documents, including images, PDFs, audio, and video files.
> 
> A key requirement is to implement a storage solution and a comprehensive search mechanism that allows users to search for any type of information related to their profile, including both textual data and multimedia files. The system should be capable of searching for similar content and retrieving relevant data efficiently.
> 
> I am seeking your expertise to determine the best approach to implement this search mechanism using Milvus and Langchain.
> 
> Thank you. arian360

can you give more specific questions about what kind of help you are looking for?
If you want more paid support, please go to zilliz.com and contact our sales. Our pipeline might be something you are looking for.
if you need community support, we will need more detailed info like what is the current blocker

**Reply by arian360** [reply]
Thank you @yhmo and @xiaofan-luan.

I am new to Milvus and have seen in the documentation that it supports input data in text format for creating embeddings. My requirement is to store both structured and unstructured data in Milvus and perform content-based searches on this data, similar to a chatbot. Could you please provide any guidelines or suggestions for implementing this solution?

**Reply by xiaofan-luan** [reply]
easiest way is to use langchain or llamaindex . they have many demo for that.
for cloud user, use zilliz cloud pipeline can directly help on that
